net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/builder/CheckerFactory.java

```diff
@@ -247,4 +247,38 @@ private static Checker createCheckerInternal(InputSource input, PropertyResolver
 
     return checker;
   }
+
+  private static void setLocaleIfChanged(final Checker checker) {
+    final String lc = getLocale();
+    if (lc != null && !lc.equals(CheckstylePlugin.getPlatformLocale().getLanguage())) {
+      setLocale(checker, lc);
+    }
+  }
+
+  private static void setLocale(final Checker checker, final String lang) {
+    final String lastLocale;
+    if (lang != null) {
+      lastLocale = lang;
+      checker.setLocaleLanguage(lang);
+      checker.setLocaleCountry("");
+      final Locale locale = new Locale(lang);
+      LocalizedMessage.setLocale(locale);
+      CheckstylePlugin.setPlatformLocale(locale);
+    } else {
+      // set the eclipse platform locale
+      final Locale platformLocale = CheckstylePlugin.getPlatformLocale();
+      lastLocale = platformLocale.getLanguage();
+      checker.setLocaleLanguage(lastLocale);
+      checker.setLocaleCountry(platformLocale.getCountry());
+      LocalizedMessage.setLocale(platformLocale);
+    }
+  }
+
+  private static String getLocale() {
+    String lang = CheckstylePluginPrefs.getString(CheckstylePluginPrefs.PREF_LOCALE_LANGUAGE);
+    if (lang != null && !lang.isEmpty() && !"default".equals(lang)) {
+      lang = null;
```

Comment:
I'm not sure I understand what's going on here. So this method will only return null, empty string or `default`?

Reply:
Sorry, it's refactor mistake, I'll fix it.